projects/plugins/jetpack/class.jetpack.php

```diff
@@ -1082,24 +1083,17 @@ public function filter_sync_callable_whitelist( $callables ) {
 
 		// Jetpack Functions.
 		$jetpack_callables = array(
-			'single_user_site'         => array( 'Jetpack', 'is_single_user_site' ),
-			'updates'                  => array( 'Jetpack', 'get_updates' ),
-			'available_jetpack_blocks' => array( 'Jetpack_Gutenberg', 'get_availability' ), // Includes both Gutenberg blocks *and* plugins.
+			'single_user_site'              => array( 'Jetpack', 'is_single_user_site' ),
+			'updates'                       => array( 'Jetpack', 'get_updates' ),
+			'available_jetpack_blocks'      => array( 'Jetpack_Gutenberg', 'get_availability' ), // Includes both Gutenberg blocks *and* plugins.
+			'sso_is_two_step_required'      => array( Helpers::class, 'is_two_step_required' ), // SSO feature related.
```

Comment:
I'm wondering if I should add a module active check here.

Reply:
Adding a module active check makes sense to me, so that we only sync those callables when the SSO module is active. 
I would also suggest moving this logic to the Sync package since those callables are no longer Jetpack plugin specific, but available to all plugins via the Connection package. 
Maybe we should also consider adding them to the [minimal Sync configuration as required ones](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/sync#jetpack_sync_callable_whitelist--jetpack_sync_multisite_callable_whitelist) so that if the SSO module is enabled, they will always get synced no matter the consumer Sync configuration. The list of required callables for Sync is defined [here](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/sync/src/class-data-settings.php#L26).
What do you think?

Reply:
Thanks for the feedback!

> I would also suggest moving this logic to the Sync package since those callables are no longer Jetpack plugin specific, but available to all plugins via the Connection package.

That makes sense. Wouldn't it be better in the Connection package directly, in the SSO class? This way the SSO module check wouldn't be necessary, and the Sync settings would be close to what it relies on / is used for.

> Maybe we should also consider adding them to the [minimal Sync configuration as required ones](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/sync#jetpack_sync_callable_whitelist--jetpack_sync_multisite_callable_whitelist) so that if the SSO module is enabled, they will always get synced no matter the consumer Sync configuration. The list of required callables for Sync is defined [here](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/sync/src/class-data-settings.php#L26).

That's a good point. At the same time, we already have one standalone (A4A) that ships without My Jetpack and is not interested in SSO as far as I know. Wouldn't we want to keep the minimum sync as small as possible for them?

Reply:
> Wouldn't it be better in the Connection package directly, in the SSO class? 

Actually yes, I think that would be the best approach 👍 And since we rely on a filter, it would be fine even in cases where we do have the Connection package but not Sync (eg WooCommerce).

> Wouldn't we want to keep the minimum sync as small as possible for them?

If we add those in the Connection package using the corresponding filter, adding those to Sync's minimal config is not applicable anymore. But even if we decided to have them in Sync, we would only add them to its minimal config when the SSO module is enabled so I believe this wouldn't be a concern for plugins like A4A

Overall I think your suggestion to add those callables in the Connection package is the right one!

Reply:
👍 Done in d05c401a6935cd8cf2477bdbf919ddc753c744a8

Reply:
hm, the Sync tests are now failing in the Jetpack plugin. Which makes sense since we now hook only when the module is active.
I suppose we could fix the tests by activating the module first, but I wonder if the tests should even stay in the Jetpack plugin, when everything else is moving out of the plugin. What's your take on this? I don't suppose I could run similar tests inside the Connection package, since Sync is not going to be easily accessible there?

Reply:
@fgiannar How do you think I should address those changes to the Sync tests?

Reply:
Given the nature of the sync tests we run in the Jetpack plugin (integration vs unit) I think it's best to leave them inside the Jetpack plugin (we even do the same for Woo related ones) and activate the module first as you suggested.

Btw apologies for the delayed reply here as I missed this one coming back from AFK.